pkg/cmd/checkendpoints/controller/connection_checker.go

```diff
@@ -25,18 +28,21 @@ type ConnectionChecker interface {
 }
 
 // NewConnectionChecker returns a ConnectionChecker.
-func NewConnectionChecker(check *operatorcontrolplanev1alpha1.PodNetworkConnectivityCheck, client v1alpha1helpers.PodNetworkConnectivityCheckClient, recorder events.Recorder) ConnectionChecker {
-	return &connectionChecker{
-		check:    check,
-		client:   client,
-		recorder: recorder,
-		stop:     make(chan interface{}),
+func NewConnectionChecker(check *operatorcontrolplanev1alpha1.PodNetworkConnectivityCheck, client v1alpha1helpers.PodNetworkConnectivityCheckClient, clientCerts []tls.Certificate, recorder events.Recorder) ConnectionChecker {
+	c := &connectionChecker{
+		check:       check,
+		client:      client,
+		clientCerts: clientCerts,
+		recorder:    recorder,
+		stop:        make(chan interface{}),
 	}
+	return c
 }
 
 type connectionChecker struct {
 	check       *operatorcontrolplanev1alpha1.PodNetworkConnectivityCheck
 	client      v1alpha1helpers.PodNetworkConnectivityCheckClient
+	clientCerts []tls.Certificate
```

Comment:
make this a function that produces tls.Certificate.  We'll probably drive it based on a lister.